### PR TITLE
Handle path build expr.

### PR DESCRIPTION
**Pull request by CPWstatic**

### CODE REVIEW

**Review by nevermore3 — APPROVED** [review]

**Review by nevermore3 — DISMISSED** [review]

File: src/visitor/FoldConstantExprVisitor.cpp [hunk 1]
```diff
@@ -296,5 +296,21 @@ Expression *FoldConstantExprVisitor::fold(Expression *expr) const {
...
+    for (size_t i = 0; i < items.size(); ++i) {
+        auto item = items[i].get();
+        item->accept(this);
+        if (!canBeFolded_) {
+            canBeFolded = false;
+            continue;
+        }
+        if (item->kind() != Expression::Kind::kConstant) {
+            expr->setItem(i, std::unique_ptr<Expression>{fold(item)});
+        }
```
Comment (jievince) [inline src/visitor/FoldConstantExprVisitor.cpp:311]: It's better to use isConstant() to check if item is a constant expression. If not, then item->accept(this)

Comment (CPWstatic) [reply src/visitor/FoldConstantExprVisitor.cpp:311]: ok.

File: src/visitor/ExprVisitorImpl.cpp [hunk 2]
```diff
@@ -125,5 +125,14 @@ void ExprVisitorImpl::visitBinaryExpr(BinaryExpression *expr) {
     }
 }
 
+void ExprVisitorImpl::visit(PathBuildExpression *expr) {
+    DCHECK(ok());
```
Comment (laura-ding) [inline src/visitor/ExprVisitorImpl.cpp:129]: it must true?

**Review by jievince — APPROVED** [review]
image

File: src/visitor/ExprVisitorImpl.cpp (hunk 2 above)
Comment (CPWstatic) [reply src/visitor/ExprVisitorImpl.cpp:129]: Yes, the father expr should not call `accept` once it is not ok.

**Review by nevermore3 — APPROVED** [review]

**Pull request merged**